site/cds_rdm/oidc.py

```diff
@@ -3,21 +3,43 @@
 from invenio_userprofiles.forms import confirm_register_form_preferences_factory
 from flask import current_app
 from werkzeug.local import LocalProxy
+from invenio_oauthclient import current_oauthclient
+from invenio_oauthclient.contrib.keycloak.handlers import get_user_info
 
 _security = LocalProxy(lambda: current_app.extensions["security"])
 
 
 def confirm_registration_form(*args, **kwargs):
     Form = confirm_register_form_preferences_factory(_security.confirm_register_form)
+
     class _Form(Form):
         password = None
         recaptcha = None
         submit = None  # defined in the template
+
     return _Form(*args, **kwargs)
 
 
-def cern_info_serializer(remote, resp, token_user_info, user_info):
+def cern_group_serializer(remote, groups, **kwargs):
+    """Serialize the groups response object."""
+    serialized_groups = []
+    # E-groups do have unique names and this name cannot be updated, therefore the name can act as an ID for invenio
+    for group_name in groups:
+        serialized_groups.append({"id": group_name, "name": group_name})
+
+    return serialized_groups
+
 
+def cern_group_handler(remote, resp, user):
+    """Retrieves groups from remote account."""
+    token_user_info, user_info = get_user_info(remote, resp)
+    groups = token_user_info.get("groups", [])
```

Comment:
as we do for other values in the [keycloak serializer](https://github.com/inveniosoftware/invenio-oauthclient/blob/master/invenio_oauthclient/contrib/keycloak/handlers.py#LL65C2-L65C63), would it make sense to check the `user_info` if the token does not contain groups? I know CERN's IdP gives them in the token, but just to be safe on changes.

Reply:
From my understanding in that serializer we do check for the information reuqired to create the user profile. Not sure what are the benefits of adding a check for the groups there?

Reply:
Is not about adding a check, is about safety for future changes. Having the groups in the OAuth token response is not standard (IIRC it was requested for the CERN Auth team to add them). Usually, they come in the user info request. Which means by "standard" they would come in `user_info`. Since this is `CDS-RDM` it okay to leave it as is (sorry, should've maked the comment as minor). My point was about doing something like:

```python
groups = token_user_info.get("groups", [])
if not groups:
    groups = user_info.get("groups", [])
```

In case the implementation changes back to standard in the future. But is ok to leave it as is.

Reply:
I created a ticket to improve